buildSrc/src/main/java/org/elasticsearch/gradle/tool/DockerUtils.java

```diff
@@ -0,0 +1,242 @@
+            lastResult = runCommand(dockerPath, "version", "--format", "{{.Server.Version}}");
+
+            if (lastResult.isSuccess() == true) {
+                version = Version.fromString(lastResult.stdout.trim(), Version.Mode.RELAXED);
+
+                isVersionHighEnough = version.onOrAfter("17.05.0");
+
+                if (isVersionHighEnough == true) {
+                    // Check that we can execute a privileged command
+                    lastResult = runCommand(dockerPath, "images");
+                }
+            }
+        }
+
+        boolean isAvailable = isVersionHighEnough && lastResult.isSuccess() == true;
+
+        return new DockerAvailability(isAvailable, isVersionHighEnough, dockerPath, version, lastResult);
+    }
+
+    /**
+     * An immutable class that represents the results of a Docker search from {@link #getDockerAvailability()}}.
+     */
+    public static class DockerAvailability {
+        /**
+         * Indicates whether Docker is available and meets the required criteria.
+         * True if, and only if, Docker is:
+         * <ul>
+         *     <li>Installed</li>
+         *     <li>Executable</li>
+         *     <li>Is at least version 17.05</li>
+         *     <li>Can execute a command that requires privileges</li>
+         * </ul>
+         */
+        public final boolean isAvailable;
+
+        /**
+         * True if the installed Docker version is &gt;= 17.05
+         */
+        public final boolean isVersionHighEnough;
+
+        /**
+         * The path to the Docker CLI, or null
+         */
+        public final String path;
+
+        /**
+         * The installed Docker version, or null
+         */
+        public final Version version;
+
+        /**
+         * Information about the last command executes while probing Docker, or null.
+         */
+        public final Result lastCommand;
+
+        public DockerAvailability(boolean isAvailable, boolean isVersionHighEnough, String path, Version version, Result lastCommand) {
+            this.isAvailable = isAvailable;
+            this.isVersionHighEnough = isVersionHighEnough;
+            this.path = path;
+            this.version = version;
+            this.lastCommand = lastCommand;
+        }
+    }
+
+    /**
+     * Given a list of tasks that requires Docker, check whether Docker is available, otherwise
+     * throw an exception.
+     * @param tasks the tasks that require Docker
+     * @throws GradleException if Docker is not available. The exception message gives the reason.
+     */
+    public static void assertDockerIsAvailable(List<String> tasks) {
+        DockerAvailability availability = null;
+
+        try {
+            availability = getDockerAvailability();
+        } catch (Exception e) {
+            throwDockerRequiredException("Something went wrong while checking Docker's availability", e);
+        }
+
+        if (availability.isAvailable == true) {
```

Comment:
Although the `== false` is a convention we use, I'm not sure `boolean == true` is conventional in the ES project. I think we can just simplify this to `if (availability.isAvailable)`.